duckduckgo/macos-browser
# Add GHA workflow to cut release branch

Task/Issue URL: https://app.asana.com/0/1203301625297703/1206152126256397/f

**Description**:
Add code_freeze.yml that takes no arguments and must be run from main branch. It evaluates
next release version number, creates a release branch, updates version and embedded files,
creates a new release task in Asana, assigns the task to the actor (person running the workflow),
runs PR checks workflow, updates version number, prepares App Store and DMG builds
and uploads artifacts to Asana task.

**Steps to test this PR**:
Check out the [sample workflow run](https://github.com/duckduckgo/macos-browser/actions/runs/7219510343) from this branch:
* [verify that new version is properly calculated by bumping minor component](https://github.com/duckduckgo/macos-browser/actions/runs/7219510343/job/19670660162#step:4:534)
* [verify that embedded files are updated](https://github.com/duckduckgo/macos-browser/actions/runs/7219510343/job/19670660162#step:4:596)
* [verify that version is bumped and commits are pushed to the release branch](https://github.com/duckduckgo/macos-browser/actions/runs/7219510343/job/19670660162#step:4:612)
* [verify that the release task is created](https://github.com/duckduckgo/macos-browser/actions/runs/7219510343/job/19670660162#step:5:1)
* verify the [task created in Asana](https://app.asana.com/0/1201037661562251/1206179017085934/f) and that Dax assigned it to the person running the workflow
* verify that PR checks forkflow is run (Run Tests job)
* [verify that the build number is increased](https://github.com/duckduckgo/macos-browser/actions/runs/7219510343/job/19670901104#step:4:80)
* verify that release build workflows are triggered on a proper (release) branch: [DMG](https://github.com/duckduckgo/macos-browser/actions/runs/7219510343/job/19670916883#step:1:42) and [AppStore](https://github.com/duckduckgo/macos-browser/actions/runs/7219510343/job/19670917145#step:1:42)
* [verify that contents of the release branch make sense](https://github.com/duckduckgo/macos-browser/compare/dominik/gha-cut-release-branch...release/1.69.0) 



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
